frame/oracle/src/lib.rs

```diff
@@ -386,29 +389,48 @@ pub mod pallet {
 		type Timestamp = <T as frame_system::Config>::BlockNumber;
 		type LocalAssets = T::LocalAssets;
 		type MaxAnswerBound = T::MaxAnswerBound;
+		type TwapWindow = T::TwapWindow;
 
 		fn get_price(
 			asset_id: Self::AssetId,
 			amount: Self::Balance,
 		) -> Result<Price<Self::Balance, Self::Timestamp>, DispatchError> {
 			let Price { price, block } =
 				Prices::<T>::try_get(asset_id).map_err(|_| Error::<T>::PriceNotFound)?;
-			let unit = 10_u128
-				.checked_pow(Self::LocalAssets::decimals(asset_id)?)
-				.ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
-			let price = multiply_by_rational(price.into(), amount.into(), unit)
-				.map_err(|_| DispatchError::Arithmetic(ArithmeticError::Overflow))?;
-			let price = price
-				.try_into()
-				.map_err(|_| DispatchError::Arithmetic(ArithmeticError::Overflow))?;
+			let price = Self::quote(asset_id, price, amount)?;
 			Ok(Price { price, block })
 		}
 
-		fn get_twap(
-			of: Self::AssetId,
-			weighting: Vec<Self::Balance>,
+		/// Currently using a flat distribution of weights.
+		fn get_twap_for_amount(
+			asset_id: Self::AssetId,
+			amount: Self::Balance,
 		) -> Result<Self::Balance, DispatchError> {
-			Self::get_twap(of, weighting)
+			let prices_length = Self::price_history(asset_id).len();
+			if prices_length == 0 {
+				Self::get_price(asset_id, amount).map(|p| p.price)
+			} else {
+				let twap_window = Self::TwapWindow::get().into();
+				let weights_length = prices_length.min(twap_window);
+				// make flat weights
+				let weight = Percent::from_rational(1, weights_length);
```

Comment:
We can just remove the weight calculation as the TWAP for flat weights is just equal to 
`(sum of prices for the window) / window_length` where `window_length = min(prices_length, twap_window_length)`. i.e no need to call `get_twap`